prov/mrail/src/mrail_ep.c

```diff
@@ -72,6 +78,23 @@ static ssize_t mrail_ep_sendmsg(struct fid_ep *ep_fid, const struct fi_msg *msg,
 	rail_msg.addr = *(fi_addr_t *)ofi_av_get_addr(mrail_ep->util_ep.av,
 						      (int)msg->addr);
 
+	if (rail_msg.desc) {
+		int i;
+		void **descs = calloc(rail_msg.iov_count, sizeof(void *));
```

Comment:
This is a memory leak, plus we really don't want memory allocations in the middle if sendmsg.